core/validator_manager.go

```diff
@@ -0,0 +1,137 @@
+package core
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+	"sync"
+
+	"github.com/0xPolygon/go-ibft/messages/proto"
+)
+
+var (
+	errVotingPowerNotCorrect = errors.New("total voting power is zero or less")
+)
+
+// ValidatorManager keeps voting power and other informations about validators
+type ValidatorManager struct {
+	vpLock *sync.RWMutex
+
+	// quorumSize represents quorum for the height specified in the current View
+	quorumSize *big.Int
+
+	// validatorsVotingPower is a map of the validator addresses on their voting power for
+	// the height specified in the current View
+	validatorsVotingPower map[string]*big.Int
+
+	log Logger
+}
+
+// NewValidatorManager creates new ValidatorManager
+func NewValidatorManager(log Logger) *ValidatorManager {
+	return &ValidatorManager{
+		quorumSize:            big.NewInt(0),
+		validatorsVotingPower: nil,
```

Comment:
You don't need this setting to `nil`.

Reply:
know, but just to sleep more peacefully  :)

Reply:
I agree with Goran, if you are about to sleep more peacefully, then I'm going to have nightmares if this remains. :slightly_smiling_face:

Reply:
Disagree, what if compiler changes in two years? Why would I think about that?
And visually is better to grasp with the code.

Reply:
> Disagree, what if compiler changes in two years?

I cannot imagine this would happen and don't accept this is a valid argument.

> And visually is better to grasp with the code.

I don't perceive that it is any easier to understand the code this way. To me, it is completely the opposite (asking myself what was author thinking)... But don't want to deepen the discussion, it is your call at the end of the day.